### PR TITLE
migrate to flutter_map 3

### PR DESCRIPTION
**Pull request by MooNag**
fleaflet/flutter_map#1333

### CODE REVIEW

File: example/lib/popup_outside_of_map.dart [hunk 1]
```diff
@@ -10,10 +10,10 @@ class PopupOutsideOfMap extends StatefulWidget {
   const PopupOutsideOfMap({Key? key}) : super(key: key);
 
   @override
-  State<PopupOutsideOfMap> createState() => _PopupOutsideOfMapState();
+  State<PopupOutsideOfMap> createState() => _PopupOutsideOfFlutterMapState();
 }
 
-class _PopupOutsideOfMapState extends State<PopupOutsideOfMap> {
+class _PopupOutsideOfFlutterMapState extends State<PopupOutsideOfMap> {
```
Comment (rorystephenson) [inline example/lib/popup_outside_of_map.dart:16]: I think this renaming is a search-replace error. _PopupOutsideOfMapState means the state of PopupOutsideOfMap.

**Comment by rorystephenson** [comment]
Thank you very much for the contribution! I made one little comment. Apart from that I noticed that a package source was changed to a github url so I guess this PR is in standby until v3 is published officially?

File: example/lib/popup_outside_of_map.dart (hunk 1 above)
Comment (MooNag) [reply example/lib/popup_outside_of_map.dart:16]: Haha I think you are right! Will fix!

**Comment by MooNag** [comment]
V3 is published and this PR is ready!

File: example/pubspec.lock [hunk 2]
```diff
@@ -291,4 +284,4 @@ packages:
     version: "2.0.0"
 sdks:
   dart: ">=2.17.1 <3.0.0"
-  flutter: ">=2.0.0"
```
Comment (rorystephenson) [inline example/pubspec.lock:294]: What causes this to jump to 3.3?

File: lib/src/popup_marker_layer_widget.dart [hunk 3]
```diff
@@ -31,33 +31,30 @@ class _PopupMarkerLayerWidgetState extends State<PopupMarkerLayerWidget> {
   Widget build(BuildContext context) {
     return PopupStateWrapper(
       builder: (context, popupState) => _layers(
-        mapState: flutter_map.MapState.maybeOf(context)!,
+        mapState: flutter_map.FlutterMapState.maybeOf(context)!,
```
Comment (rorystephenson) [inline lib/src/popup_marker_layer_widget.dart:34]: Maybe now that `MapState` is renamed to `FlutterMapState` we don't need to disambiguate with flutter_map.FlutterMapState?

**Review by rorystephenson — COMMENTED** [review]
Some quick small comments but it's looking almost ready!

File: lib/src/popup_marker_layer_widget.dart (hunk 3 above)
Comment (MooNag) [reply lib/src/popup_marker_layer_widget.dart:34]: Still need to with your custom MarkerLayer class.

File: example/pubspec.lock (hunk 2 above)
Comment (MooNag) [reply example/pubspec.lock:294]: Flutter 3.3 is the latest stable release as well as being required by flutter_map due to having completed the migration to the new image provider API.

File: lib/src/popup_marker_layer_widget.dart (hunk 3 above)
Comment (MooNag) [reply lib/src/popup_marker_layer_widget.dart:34]: Unless we remove the custom MarkerLayer Widget or change its name we need to keep the disambiguation.

**Comment by rorystephenson** [comment]
Awesome, thanks for the contribution @MooNag !

**Pull request merged**